packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -144,6 +144,11 @@ export async function handleRoute(
 		clientAddress: buildingToSSR ? req.socket.remoteAddress : undefined,
 	});
 
+	// Set user specified headers to response object.
+	for (const [name, value] of Object.entries(config.server.headers ?? {})) {
+		if (value) res.setHeader(name, value);
+	}
+
```

Comment:
I'm not sure why but passing `server.headers` to `vite` config didn't work. Hence, writing the headers here directly.